### DISCUSSION TITLE
Controlling more than one heat pump

### DISCUSSION THREAD

**Discussion by Holric** [edterbak/HeishaMoNR#54] [Q&A]
Hello,
thank you for providing the Heishamon control!
I have two 5kW heat pumps in a cascade, both have a HeishaMon connected. Would it be be possible to control more than one unit via NodeRed Heishamon control?

**Comment by edterbak** [comment]
HI!! :) 

Sorry for the late reply. 
I just noticed a new discussion. I do not get notifications I see now.. (need to change that)

I guess it should be possible.
But I am wondering first how you have things connected. (not wired)
you have 2 separate Heishamon units. Both have their own IP adress I would guess.
But do both units send their measurements to 1 MQTT broker? 
How are the two units recognized in mqt topics. 

What I can imagine that would work.
Unit 1 >> heishamon 1 >> broker 1:
Unit 2 >> heishamon 2 >> broker 2:

You also require 2 instances of node red. This is because of the programmed variables coming from incoming heishamon data. The flow is not aware of more than 2 source. 
So if you have 2 separate node red instances. you can control both on theirh own nodered dashboard.

Sounds a bit over the top if you are thinking of using a raspberry pi 
But if you are running something like proxmox, one can easily make a second node red container and a second mqtt broker. 


Fun question :)

To round off.
For this flow to control 2 devices simultaneaous, it needs be rewritten completly.
